consensus/istanbul/backend/engine.go

```diff
@@ -517,8 +517,7 @@ func (sb *backend) Finalize(chain consensus.ChainReader, header *types.Header, s
 	// If sb.chain is nil, it means backend is not initialized yet.
 	if sb.chain != nil && !reward.IsRewardSimple(pset) {
 		// TODO-Kaia Let's redesign below logic and remove dependency between block reward and istanbul consensus.
-
-		lastHeader := chain.CurrentHeader()
+		lastHeader := chain.GetHeader(header.ParentHash, header.Number.Uint64()-1)
 		valSet := sb.getValidators(lastHeader.Number.Uint64(), lastHeader.Hash())
 
```

Comment:
Maybe we can move these 2 lines inside the if statement because they're only used inside the if.

Reply:
Applied at [5405c12](https://github.com/kaiachain/kaia/pull/48/commits/5405c12a3c55abbf0881242ca35c47d67621e11d)